lib/krb5/principal.c

```diff
@@ -205,200 +206,190 @@ krb5_principal_get_num_comp(krb5_context context,
  * @ingroup krb5_principal
  */
 
-KRB5_LIB_FUNCTION krb5_error_code KRB5_LIB_CALL
-krb5_parse_name_flags(krb5_context context,
-		      const char *name,
-		      int flags,
-		      krb5_principal *principal)
+
+#define IN_REALM	0x01
+#define	FIRST_AT	0x02
+
+#ifndef MIN
+#define MIN(x,y)	((x) < (y)) ? (x) : (y)
+#endif
+
+static krb5_error_code
+add_bit(krb5_context ctx, krb5_principal *p, int *state, char sep,
```

Comment:
Perhaps a brief note about what this function is supposed to do. The name is not very evocative...